proposed/2022/package-scoring-v1.md

```diff
@@ -0,0 +1,222 @@
+
+- Package updated in the last year.
+
+The score may also be shown as a decayed value over the year approaching the year mark where it goes stale.
+
+Total Score - 100
+
+#### Security
+
+Finally, Security represents the trust indicators representing the package in the current state of the supply chain risk. This includes known vulnerabilities, license risk, and any issues surrounding secure supply chain risk.
+
+- No known security vulnerabilities (Critical, High, Moderate, Low) in top-level or transitive dependencies.
+- Specifies a valid license (SPDX, embedded)
+- Not deprecated.
+
+An emphasis would be placed on known security vulnerabilities and license restriction.
+
+Total Score - 100
+
+### Technical explanation
+
+<!-- Explain the proposal in sufficient detail with implementation details, interaction models, and clarification of corner cases. -->
+
+## Drawbacks
+
+<!-- Why should we not do this? -->
+- This is a novel area and hasn't been done from the security perspective by any major package ecosystem. There are working groups trying to solve this problem today.
+- This can paint packages in an unfair light for issues they may not be able to actually control but perceived that they can.
+- This can be gamified to a certain degree.
+- This is one model of package issues and scores. There are other models such as all-up scorecards based on best practices.
+
+## Rationale and alternatives
+
+<!-- Why is this the best design compared to other designs? -->
+<!-- What other designs have been considered and why weren't they chosen? -->
+<!-- What is the impact of not doing this? -->
+
+As mentioned in the [original spec](https://github.com/dotnet/designs/pull/216), there is a significant value add to user needs from regular surveys with regards to:
+
+- It's hard to tell if a package is of high quality and actively maintained. (5.27 Score)
+- It has insufficient package documentation (i.e. Readme, changelog, examples, API reference). (4.81 Score)
+- It is hard to tell if a package is secure. (4.61 Score)
+
+In addition in recent surveys of 2022, we have seen other themes pop up as well such as:
+
+- Clear metrics that help me evaluate package quality (3rd most popular averaging 16/100 points spent on this area)
+- Special badges for key ecosystem projects, trusted packages that significantly contribute to the .NET ecosystem  (4th most popular averaging 14/100 points spent on this area)
+
+We then asked how people install a package with the following emphasis in order.
+
+1. Package solves my problem.
+2. Package has enough downloads.
+3. Package is open source.
+4. Package has high quality documentation.
+5. Package is maintained by a notable author or organization.
+6. Package has been updated recently and is updated regularly.
+7. Package has few dependencies.
```

Comment:
I wonder if it's worth analyzing or surfacing the total size of a package's dependency graph or just consider the top level. If the later, maybe a score based on this could be gamed/unintentionally obfuscated by having a single meta-package dependency.